src/header-generator.js

```diff
@@ -0,0 +1,310 @@
+ * @param {string} httpVersion - Http version to be used to generate headers (the headers differ depending on the version).
+ *  Either 1 or 2. If none specified the httpVersion specified in `HeaderGeneratorOptions` is used.
+ */
+/**
+ * @typedef HeaderGeneratorOptions
+ * @param {Array<BrowserSpecification>} browsers - List of BrowserSpecifications to generate the headers for.
+ * @param {Array<string>} operatingSystems - List of operating systems to generate the headers for.
+ *  The options are `windows`, `macos`, `linux`, `android` and `ios`.
+ * @param {Array<string>} devices - List of devices to generate the headers for. Options are `desktop` and `mobile`.
+ * @param {Array<string>} locales - List of at most 10 languages to include in the
+ *  [Accept-Language](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language) request header
+ *  in the language format accepted by that header, for example `en`, `en-US` or `de`.
+ * @param {string} httpVersion - Http version to be used to generate headers (the headers differ depending on the version).
+ *  Can be either 1 or 2. Default value is 2.
+ */
+
+/**
+ * HeaderGenerator randomly generates realistic browser headers based on specified options.
+ */
+class HeaderGenerator {
+    /**
+     * @param {HeaderGeneratorOptions} options - default header generation options used unless overridden
+     */
+    constructor(options = {}) {
+        ow(options, 'HeaderGeneratorOptions', ow.object.exactShape(headerGeneratorOptionsShape));
+        this.defaultOptions = options;
+        this.headersOrder = JSON.parse(fs.readFileSync(headersOrderFilePath, { encoding: 'utf8' }));
+        const uniqueBrowserStrings = JSON.parse(fs.readFileSync(browserHelperFilePath, { encoding: 'utf8' }));
+        this.uniqueBrowsers = [];
+        for (const browserString of uniqueBrowserStrings) {
+            if (browserString === missingValueDatasetToken) {
+                this.uniqueBrowsers.push({
+                    name: missingValueDatasetToken,
+                });
+            } else {
+                this.uniqueBrowsers.push(prepareHttpBrowserObject(browserString));
+            }
+        }
+        this.inputGeneratorNetwork = new BayesianNetwork(inputNetworkDefinitionPath);
+        this.headerGeneratorNetwork = new BayesianNetwork(headerNetworkDefinitionPath);
+    }
+
+    /**
+     * Generates a single set of headers using a combination of the default options specified in the constructor
+     * and their possible overrides provided here.
+     * @param {HeaderGeneratorOptions} options - specifies options that should be overridden for this one call
+     */
+    getHeaders(options = {}) {
+        ow(options, 'HeaderGeneratorOptions', ow.object.exactShape(headerGeneratorOptionsShape));
+        const headerOptions = { ...this.defaultOptions, ...options };
+
+        // Set up defaults
+        if (!headerOptions.locales) {
+            headerOptions.locales = ['en-US'];
+        }
```

Comment:
Would it make sense to setup all those defaults in the constructor already? Or at least use a function `setDefaultOptions` or smth